src/main.rs

```diff
@@ -1255,19 +1263,33 @@ fn hid_recorder() -> Result<()> {
         data[0..4].copy_from_slice(&bytes_hid_id);
     }
 
-    let skel = open_skel.load().unwrap();
+    Ok(Some(open_skel.load().unwrap()))
+}
 
-    let maps = skel.maps();
+fn hid_recorder() -> Result<()> {
+    let cli = Cli::parse();
 
-    // We need to keep _link around or the program gets immediately removed
-    let _link = maps.hid_record().attach_struct_ops()?;
+    let _ = Outfile::init(&cli);
+
+    let path = match cli.path {
+        Some(path) => path,
+        None => find_device()?,
+    };
 
     let rdesc_file = find_rdesc(&path)?;
     let opts = Options { full: cli.full };
 
     let rdesc = parse(&rdesc_file, &opts)?;
     if !cli.only_describe && path.starts_with("/dev") {
-        read_events(&path, &rdesc, maps.events())?
+        match preload_bpf_tracer(cli.bpf, &path)? {
```

Comment:
if there's an error during the BPF loading (which AIUI can happen legitimately if HID-BPF is disabled) then we just bail out here instead of handling the `Result` and continuing without BPF.

Reply:
If we try to load the bpf, it's either because there is already a bpf loaded, or if the user explicitely request the loading of the bpf. In both cases it seems appropriate to bail out and not continue.